File: WheelsCrawler.Data/Helpers/UserParams.cs
namespace WheelsCrawler.Data.Helpers
{
    public class UserParams
    {
        private const int MaxPageSize = 50;
        public int PageNumber { get; set; } = 1;
        private int _pageSize = 21;
        public int PageSize
        {
            get { return _pageSize; }
            set { _pageSize = (value > MaxPageSize) ? MaxPageSize : value; }
        }

        public string City { get; set; }
        public double EngineCapacityFrom { get; set; } = 0.0;
        public double EngineCapacityTo { get; set; } = 10.0;
        public int PriceFrom { get; set; } = 0;
        public int PriceTo { get; set; } = 1_000_000;
        public int KilometrageFrom { get; set; } = 0;
        public int KilometrageTo { get; set; } = 1_000_000;
        public string OrderBy { get; set; } = "lastAdded";
        
    }
}

File: wheelscrawler-spa/src/app/models/userParams.ts
export class UserParams {
  city: string = '';
  engineCapacityFrom: number = 0.0;
  engineCapacityTo = 10.0;
  priceFrom = 0;
  priceTo = 1_000_000;
  kilometrageFrom = 0;
  kilometrageTo = 1_000_000;
  pageNumber = 1;
  pageSize = 21;

  orderBy = 'lastAdded';

  constructor() {}
}
